include/l2/logical_channel.hpp

```diff
@@ -18,6 +18,21 @@ enum class LogicalChannel {
     kStealingChannel
 };
 
+constexpr auto to_string(LogicalChannel channel) noexcept -> const char* {
```

Comment:
Why not turn this into an interface?
```c++
class Serialize {
public:
    virtual auto to_string(LogicalChannel channel) noexcept -> const char* = 0;
}
```

?

Reply:
as discussed. lets leave it like it is to not lose the constexpr